### ISSUE TITLE
BUG: Sensors sending NULL

### ISSUE THREAD

**Issue by drjakob** [roschaefer/story.board#84]
reporter.box sending null or nil or no value produces error 422 in webhook. What should reporter.box send instead? ("-1" is working but not the ideal solution.)

**Comment by roschaefer** [comment]
How about not sending invalid measurements at all? I will look into this and probably create a cucumber feature. But what's the point of sending `null` values actually?

**Comment by drjakob** [comment]
Sometimes just one value is null. E.g. before sensor calibration or if calibrated value fails. So I think solution (c) would be best.

**Comment by kabel42** [comment]
not sending invalid Measurements could be considered silently dropping errors...

**Comment by roschaefer** [comment]
@kabel42 Yes! The database of the story.board should not keep invalid data. It is not the purpose of the database to log errors. That's what the `log/production.log` are for. Is there any reason why we should save a `Sensor::Reading` with a `null` value for `calibrated_value` or `uncalibrated_value`? Is it possible for the calibration to fail and shall we save the `uncalibrated_value` then? I must admit that I strongly favour the current behaviour as we don't have to cope with `null` values in the backend then. Makes things a lot easier.

**Comment by kabel42** [comment]
If a Sensor is not yet calibrated (by the backend?), it can only send the uncalibrated_value.
The Sound-Sensor has a biny input, should it send the same Data for calibrated and uncalibrated data?
What is the current intended behaviour?
What should the backend do if the calibration changes? (recalculate all values?)

**Comment by drjakob** [comment]
I think @kabel42 ist right: we planed to do calibration by backend in the future. So a NULL value could be detection for a sensor needing to be calibrated.
